test/distributed/test_c10d.py

```diff
@@ -3815,6 +3815,38 @@ def _test_default_ddp_comm_hooks_nccl(self, gradient_as_bucket_view=False):
             # check whether the grads are equal to what DDP without hook would return.
             self._run_and_verify_hook(gpu_model, 8, 0.25 * torch.ones(2, 2))
 
+
+    def _test_default_ddp_comm_hooks_fp16_wrapper(self, gradient_as_bucket_view=False):
+        """
+        This unit test verifies whether wrapping the ALLREDUCE and POWER_SGD hooks with
+        the FP16_WRAPPER can give the same result as when there is no hook registered.
+        """
+        store = c10d.FileStore(self.file_name, self.world_size)
+        process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+        state = process_group
+
+        # test that fp16_compress_hook works with allreduce_hook.
+        gpu_model = self._gpu_model_with_ddp_comm_hook(
+            process_group,
+            default.fp16_compress_wrapper(default.allreduce_hook),
+            gradient_as_bucket_view,
+            state
+        )
+        # check whether the grads are equal to what DDP without hook would return.
+        self._run_and_verify_hook(gpu_model, 8, 0.25 * torch.ones(2, 2))
+
+        # test that fp16_compress_hook works with powerSGD comm hook.
```

Comment:
You don't need to write this much boilerplate code. Instead, you can create a loop over both `allreduce_hook` and `powerSGD_hook`.

Reply:
Yeah I thought about that -- but given that they require different states, I thought it would be more clear just to repeat. 

But I don't have any strong feelings either way, so I'll change.